.gitignore

```diff
@@ -0,0 +1,45 @@
+# Created by https://www.toptal.com/developers/gitignore/api/rust,visualstudiocode,rust-analyzer
+# Edit at https://www.toptal.com/developers/gitignore?templates=rust,visualstudiocode,rust-analyzer
+
+### Rust ###
+# Generated by Cargo
+# will have compiled files and executables
+debug/
+target/
+
+# Remove Cargo.lock from gitignore if creating an executable, leave it for libraries
+# More information here https://doc.rust-lang.org/cargo/guide/cargo-toml-vs-cargo-lock.html
+Cargo.lock
```

Comment:
I'd like to not ignore the Cargo.lock for examples, so maybe you can create a specific .gitignore for the host/ folder?